contracts/consensus/Consensus.sol

```diff
@@ -1058,16 +1166,49 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
     {
         bytes memory committeeSetupData = abi.encodeWithSelector(
             COMMITTEE_SETUP_CALLPREFIX,
+            _metachainId,
             address(this),
             _committeeSize,
             _dislocation,
             _proposal
         );
 
-        address committeeAddress = axiom.newCommittee(
-            committeeSetupData
+        committee_ = CommitteeI(axiom.newCommittee(committeeSetupData));
+    }
+
+    /**
+     * @notice hashBlockSegment() function calculates a seed based on the
+     *         specified blocksengment.
+     *
+     * @param end The ending block number of the blocksegment (included).
+     * @param length Length of the blocksegment.
+     *
+     * @return Returns a seed based on the blockhashes of the given blocksegment.
+     */
+    function hashBlockSegment(
+        uint256 end,
+        uint256 length
+    )
+        private
+        view
+        returns (bytes32 seed_)
+    {
+        assert(length > 0);
+
+        uint256 begin = end.add(uint256(1)).sub(length);
+
+        require(
+            block.number >= end && block.number < begin.add(uint256(256)),
```

Comment:
`block.number` must be strictly greater than `end`, bc the blockhash of the current block is not yet known

Reply:
Yes. 🤦. Thanks!